### DISCUSSION TITLE
Attach a file to a multipart request

### DISCUSSION THREAD

**Discussion by meleyal** [jreyesr/insomnia-plugin-batch-requests#14] [Ideas]
Would it be possible to take a local file path and use this to attach a file to a multipart/form-data request?

**Comment by jreyesr** [comment]
Hello @meleyal, and thanks for suggesting this! I'm really not sure, but it... _should_ be possible? 

Just to confirm, you're imagining a CSV file that looks like this, correct?

|path|description|
|---|---|
|/home/meleyal/Pictures/cat.jpg|Fluffy, the Bald Cat|
|/home/meleyal/Pictures/dog.jpg|Brutus, the Friendly Dog|

And then you'd configure the Insomnia request like this, right?

![image](https://github.com/jreyesr/insomnia-plugin-batch-requests/assets/23390438/ad8bcc6d-6777-424c-9957-3fca4aa3dd69)

I believe that it would be necessary to create another tag, like there's the Batch tag now. This new tag would behave like the current Batch tag, but it would have to dynamically replace itself with the _contents_ of the file whose path is contained in the CSV cell.

I'll get working on it when I get some time

**Comment by meleyal** [comment]
> Just to confirm, you're imagining a CSV file that looks like this, correct?

@jreyesr yes, exactly!

**Comment by jreyesr** [comment]
@meleyal I've released `v1.5.0`, which contains a new template tag, called `Batch (File)`. You'd configure it so it points to the desired CSV column, and then run the Batch Requests as normal. For example, on a CSV like this:

![image](https://github.com/jreyesr/insomnia-plugin-batch-requests/assets/23390438/4e539dcf-96b9-4c41-a165-7b4d3ea9242e)

you'd configure the tag like this (notice that the **Function to Perform**, above, is `Batch (File)`, not just `Batch`):

![image](https://github.com/jreyesr/insomnia-plugin-batch-requests/assets/23390438/1c9926c8-4329-4a81-b739-c883d1c7c886)

The request may look like this:

![image](https://github.com/jreyesr/insomnia-plugin-batch-requests/assets/23390438/b8ab961d-26a2-4263-9a3e-1509d940a8ab)

Since that template tag is configured to point to the `fname` CSV column, when you run the Batch Requests it'll read the contents of that column and treat them as paths. It'll then reconfigure the field in which it was included (`file` here) so it isn't a text field but a file field. The requests that are sent will be equivalent to those that would be sent if you manually chose a file on Insomnia:

![image](https://github.com/jreyesr/insomnia-plugin-batch-requests/assets/23390438/fe56816e-2db2-47ec-86d0-d6a15bcafa46)

There's more info here: https://github.com/jreyesr/insomnia-plugin-batch-requests?tab=readme-ov-file#sending-files

Feel free to experiment with the new version, and report any problems that you find. I _think_ you can update an installed plugin by trying to "install" it on `Application>Preferences>Plugins` menu, or if that doesn't work you can try uninstalling and reinstalling it.

**Comment by meleyal** [comment]
@jreyesr thanks, this works great 👍
